perf(parser): use SmallVec for duplicate default export detection

camc314 opened the pull request:
Avoid heap allocation for the common case (0-2 default exports) by using
SmallVec with inline capacity of 2 instead of Vec.

🤖 generated with help from Claude Opus 4.5

Boshen reviewed (commented):
`default_exports` is not used elsewhere, why not `.count()` 🤔 ?

camc314 committed:
perf(parser): use SmallVec for duplicate default export detection (#16801)

Avoid heap allocation for the common case (0-2 default exports) by using
SmallVec with inline capacity of 2 instead of Vec.

🤖 generated with help from Claude Opus 4.5
